### DISCUSSION TITLE
cannot import name 'ImageClassification' from 'autogluon'

### DISCUSSION THREAD

**Discussion by ghost** [autogluon/autogluon#970] [Q&A] [answered]
Hi,

I've just install mxnet/gluoncv/autogluon from sources, import of mxnet/gluoncv/autogluon raises no error but 
from autogluon import ImageClassification as task does raise the following error : 
   from autogluon import ImageClassification as task
ImportError: cannot import name 'ImageClassification' from 'autogluon' (/home/penta/autogluon/autogluon/src/autogluon/__init__.py)

Any idea how to fix that ?
Regards.

**Comment by Innixma** [comment, marked as answer]
I believe it is using pre-release of AutoGluon, which has a different API. Refer to dev documentation:

https://auto.gluon.ai/dev/tutorials/image_prediction/beginner.html

**Reply by Innixma** [reply]
Also, I believe gluoncv might not install vision module of AG. For that, you would need to do `pip install --pre autogluon` or `pip install autogluon.vision`

**Reply by ghost** [reply]
Okay so If I'm right : 
"from autogluon.vision import ImageClassification as task" is deprecated and should  be replaced by 
"import autogluon.core as agfrom autogluon.vision import ImagePredictor  as task"

**Reply by Innixma** [reply]
You'd want to simply do

`from autogluon.vision import ImagePredictor`

`predictor = ImagePredictor().fit(...)`

It would be best to follow the tutorials to understand the new API.

**Reply by Emekaborisama** [reply]
same error

`ImportError: cannot import name 'stringify' from 'dask.utils' (/usr/local/lib/python3.7/dist-packages/dask/utils.py)`
